### DISCUSSION TITLE
Explaination of ENOMSG in k_msgq_put

### DISCUSSION THREAD

**Discussion by ClaytonNorthey92** [zephyrproject-rtos/zephyr#47873] [Q&A]
Looking at the [k_msgq_put documentation](https://docs.zephyrproject.org/3.0.0/reference/kernel/data_passing/message_queues.html#c.k_msgq_put), I notice that the following error can be returned with the accompanying definition:
> -ENOMSG – Returned without waiting or queue purged. 

What does "Returned without waiting" mean?  Does that mean the message was handled without it waiting in the queue?

**Comment by JordanYates** [comment]
Because message queues can only hold a fixed number of messages, it is possible that the queue is already full when you call `k_msgq_put`.
If this is the case, but you also specified not to wait at all (via `K_NO_WAIT`), `-ENOMSG` is returned.
https://github.com/zephyrproject-rtos/zephyr/blob/62f19cc6b3d4697630abed9c5bea94d3a4eed7ed/kernel/msg_q.c#L155-L158
